keps/sig-cloud-provider/20190215-promoting-cloud-provider-labels.md

```diff
@@ -0,0 +1,178 @@
+promote these labels from beta to GA.
+
+### Goals
+
+* promote cloud provider node/volume labels to GA with minimal visible changes to users.
+* remove the usage of "beta" cloud provider node/volume labels without breaking compatiblity guaranetees. This will span many Kubernetes versions as per the Kubernetes deprecation policy.
+
+### Non-Goals
+
+* introducing more labels
+* changing the behaviour of these labels within the Kubernetes system.
+
+## Proposal
+
+In order to promote these labels to GA safely, there will be a period in which both the "beta" and "GA" labels are applied to node and volume objects.
+This is required in order to maintain backwards compatiblity as many clusters rely on the beta labels today.
+
+Both new and existing resources will require both the beta and GA versions of the labels. For the case of existing resources, keeping the beta labels is a requirement in order for
+existing workloads to behave as expected. For the case of new resources, both labels are still requried as workloads may still consume the beta labels in some other resource
+that was not updated yet to use the GA labels. One possible example is where a deployment may still use the beta zone label (`failure-domain.beta.kubernetes.io/zone`) as a
+nodeSelector and not applying the beta labels to new nodes would mean new nodes in that zone would not be considered when pods are being scheduled.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Here is a break down of the implementation steps:
+
+1) [v1.14] update components to apply both the GA and beta labels to nodes & volumes.
+2) [v1.14] deprecate the beta labels.
+3) [v1.14] update the appropriate release notes & documentation to promote the use of GA labels over beta labels.
+4) [v1.15] continue to promote usage of GA labels over beta labels.
+5) [v1.16] continue to promote usage of GA labels over beta labels.
+6) [v1.17] continue to promote usage of GA labels over beta labels.
+7) [v1.18] stop applying beta labels to new resources, existing resources will continue to have those labels unless manually removed.
+
+### Risks and Mitigations
+
+* duplicate labels that do the same thing can be confusing/annoying for users
+* post v1.18 Kubernetes clusters may have danging labels that provide no function
+* improper handling of labels can lead to critical bugs in scheduling / volume topology / node registration / etc.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Examples
+
+##### Alpha -> Beta Graduation
+
+TBD
+
+##### Beta -> GA Graduation
+
+TBD
+
+**Note:** Generally we also wait at least 2 releases between beta and GA/stable, since there's no opportunity for user feedback, or even bug reports, in back-to-back releases.
+
+### Upgrade / Downgrade Strategy
```

Comment:
There were a lot more edge cases here than I had originally thought. Happy to list them out in more detail once we agree to move forward with this.